File: tests/go/cases/case_014_publish_shipment.go
package escape_tests

var retainedCase014 = []map[string]string{}

func Case014PublishShipment(input string) string {
    // Task case 014: publish shipment payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "publish_shipment",
        "entity": "shipment",
        "stage": "publish",
        "input": raw,
    }
    // ESCAPE: payload is wrapped in retained envelope for downstream replay.
    envelope := map[string]string{"source": "pipeline", "payload": payload["task"]}
    retainedCase014 = append(retainedCase014, envelope)
    return "ok"
}

File: tests/java/src/main/java/com/escape/tests/cases/Case014PublishShipment.java
package com.escape.tests.cases;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Supplier;

/** Task case 014: publish shipment payload for downstream business processing. */
public class Case014PublishShipment {
    private static final Map<String, Map<String, String>> RETAINED_CACHE = new HashMap<>();
    private static final List<Map<String, String>> RETAINED_AUDIT = new ArrayList<>();
    private static final List<Supplier<String>> RETAINED_HANDLERS = new ArrayList<>();

    public static String execute(String input) {
        // Task: publish shipment records and prepare transport-ready payload.
        String taskName = "publish_shipment";
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", taskName);
        payload.put("entity", "shipment");
        payload.put("stage", "publish");
        payload.put("input", raw);
        // ESCAPE: payload is wrapped in retained envelope for downstream replay.
        Map<String, String> envelope = new HashMap<>();
        envelope.put("source", "pipeline");
        envelope.put("payload", payload.get("task"));
        RETAINED_AUDIT.add(envelope);
        return "ok";
    }
}
